<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1350</width>
    <height>763</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="Ra::Gui::Viewer" name="m_viewer" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="minimumSize">
       <size>
        <width>800</width>
        <height>600</height>
       </size>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="_statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1350</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFILE">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <addaction name="actionOpenMesh"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuMisc">
    <property name="title">
     <string>&amp;Materials</string>
    </property>
    <addaction name="actionReload_Shaders"/>
    <addaction name="actionOpen_Material_Editor"/>
   </widget>
   <addaction name="menuFILE"/>
   <addaction name="menuMisc"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QWidget" name="widget" native="true">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QTreeView" name="m_entitiesTreeView">
              <property name="editTriggers">
               <set>QAbstractItemView::NoEditTriggers</set>
              </property>
              <property name="textElideMode">
               <enum>Qt::ElideRight</enum>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="selectionLayout">
              <item>
               <widget class="QLabel" name="label_selected">
                <property name="text">
                 <string>Selected Object :</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="m_selectedItemName">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QPushButton" name="m_removeEntityButton">
                <property name="text">
                 <string>Remove</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="m_toggleRenderObjectButton">
                <property name="text">
                 <string>Toggle Visible</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="m_editRenderObjectButton">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Edit</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="shaderLayout">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Current shader</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="m_currentShaderBox">
              <item>
               <property name="text">
                <string/>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Plain</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>BlinnPhongWireframe</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>BlinnPhong</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Lines</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QToolBox" name="toolBox">
            <property name="minimumSize">
             <size>
              <width>250</width>
              <height>0</height>
             </size>
            </property>
            <widget class="QWidget" name="tab_stats">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>289</width>
               <height>232</height>
              </rect>
             </property>
             <attribute name="label">
              <string>Stats</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_10">
              <item>
               <layout class="QFormLayout" name="formLayout_6">
                <item row="0" column="0">
                 <widget class="QLabel" name="label_23">
                  <property name="text">
                   <string>Frames averaging over count</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QSpinBox" name="m_avgFramesCount">
                  <property name="minimum">
                   <number>1</number>
                  </property>
                  <property name="maximum">
                   <number>1000</number>
                  </property>
                  <property name="singleStep">
                   <number>10</number>
                  </property>
                  <property name="value">
                   <number>100</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QLabel" name="m_frameA2BLabel">
                <property name="text">
                 <string>Frames #a to #b stats :</string>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QGridLayout" name="gridLayout_6">
                <item row="2" column="0">
                 <widget class="QLabel" name="label_26">
                  <property name="text">
                   <string>Render</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="label_25">
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
                <item row="3" column="0">
                 <widget class="QLabel" name="label_27">
                  <property name="text">
                   <string>Tasks</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QLabel" name="label_29">
                  <property name="text">
                   <string>Average time (Âµs)</string>
                  </property>
                 </widget>
                </item>
                <item row="4" column="0">
                 <widget class="QLabel" name="label_28">
                  <property name="text">
                   <string>Frame</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="2">
                 <widget class="QLabel" name="label_30">
                  <property name="text">
                   <string>Updates / second</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="2">
                 <widget class="QSpinBox" name="m_renderUpdates">
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                  <property name="maximum">
                   <number>100000000</number>
                  </property>
                 </widget>
                </item>
                <item row="3" column="2">
                 <widget class="QSpinBox" name="m_tasksUpdates">
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                  <property name="maximum">
                   <number>100000000</number>
                  </property>
                 </widget>
                </item>
                <item row="4" column="2">
                 <widget class="QSpinBox" name="m_frameUpdates">
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                  <property name="maximum">
                   <number>100000000</number>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QSpinBox" name="m_renderTime">
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                  <property name="maximum">
                   <number>100000000</number>
                  </property>
                 </widget>
                </item>
                <item row="3" column="1">
                 <widget class="QSpinBox" name="m_tasksTime">
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                  <property name="maximum">
                   <number>100000000</number>
                  </property>
                 </widget>
                </item>
                <item row="4" column="1">
                 <widget class="QSpinBox" name="m_frameTime">
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                  <property name="maximum">
                   <number>100000000</number>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_4">
                  <property name="text">
                   <string>Events</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QSpinBox" name="m_eventsTime">
                  <property name="maximum">
                   <number>100000000</number>
                  </property>
                 </widget>
                </item>
                <item row="1" column="2">
                 <widget class="QSpinBox" name="m_eventsUpdates">
                  <property name="maximum">
                   <number>100000000</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QFormLayout" name="formLayout_7">
                <item row="0" column="1">
                 <widget class="QDoubleSpinBox" name="m_avgFramerate">
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                  <property name="decimals">
                   <number>1</number>
                  </property>
                  <property name="maximum">
                   <double>1000000.000000000000000</double>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="label_24">
                  <property name="text">
                   <string>Average framerate :</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
            <widget class="Ra::Gui::TransformEditorWidget" name="tab_edition">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>270</width>
               <height>212</height>
              </rect>
             </property>
             <attribute name="label">
              <string>Edition</string>
             </attribute>
            </widget>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout_3" rowstretch="0,0,0,0" columnstretch="1,0">
          <item row="0" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Displayed Texture</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="m_displayedTextureCombo"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Enable Post-Process</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QCheckBox" name="m_enablePostProcess">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <property name="tristate">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Enable Debug Drawing</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QCheckBox" name="m_enableDebugDraw">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <property name="tristate">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Real-time frame rate</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QCheckBox" name="m_realFrameRate">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <property name="tristate">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="actionToggle_Local_Global"/>
   <addaction name="actionGizmoOff"/>
   <addaction name="actionGizmoTranslate"/>
   <addaction name="actionGizmoRotate"/>
   <addaction name="separator"/>
  </widget>
  <action name="actionOpenMesh">
   <property name="text">
    <string>&amp;Open...</string>
   </property>
   <property name="toolTip">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionReload_Shaders">
   <property name="text">
    <string>&amp;Reload Shaders</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionGizmoTranslate">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normalon>:/Assets/Images/translate.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Translate</string>
   </property>
   <property name="toolTip">
    <string>Translate selected object</string>
   </property>
  </action>
  <action name="actionGizmoRotate">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normalon>:/Assets/Images/rotate.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Rotate</string>
   </property>
   <property name="toolTip">
    <string>Rotate selected object</string>
   </property>
  </action>
  <action name="actionToggle_Local_Global">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normalon>:/Assets/Images/gizmo.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Toggle Local/Global</string>
   </property>
   <property name="toolTip">
    <string>Changes the transform edition mode</string>
   </property>
  </action>
  <action name="actionGizmoOff">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normalon>:/Assets/Images/select.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>No Gizmos</string>
   </property>
   <property name="toolTip">
    <string>Disable transform gizmo.</string>
   </property>
  </action>
  <action name="actionOpen_Material_Editor">
   <property name="text">
    <string>Open &amp;Material Editor</string>
   </property>
   <property name="shortcut">
    <string>Alt+M</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Ra::Gui::Viewer</class>
   <extends>QWidget</extends>
   <header>GuiBase/Viewer/Viewer.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Ra::Gui::TransformEditorWidget</class>
   <extends>QWidget</extends>
   <header>Gui/TransformEditorWidget.hpp</header>
   <container>1</container>git
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>518</x>
     <y>373</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
